### PR TITLE
Improve fix version selection logic

### PR DESCRIPTION
**Pull request by omerzi**
- [x] All [tests](https://github.com/jfrog/frogbot#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.

---
Fixes https://github.com/jfrog/frogbot/issues/117

To avoid breakage, Frogbot upgrades to the minimum fix version. The maximum version will be chosen from all these minimum versions, which will fix all the vulnerabilities.

### CODE REVIEW

File: commands/createfixpullrequests.go [hunk 1]
```diff
@@ -295,7 +296,8 @@ func NewFixVersionInfo(newFixVersion string, packageType PackageType) *FixVersio
 }
 
 func (fvi *FixVersionInfo) UpdateFixVersion(newFixVersion string) {
-	if fvi.fixVersion == "" || version.NewVersion(fvi.fixVersion).AtLeast(newFixVersion) {
+	// fvi.fixVersion < newFixVersion => fvi.fixVersion = newFixVersion
```
Comment (sverdlov93) [inline commands/createfixpullrequests.go:299]:
```suggestion
	// Update fvi.fixVersion as the maximum version if found a new version that greater than the previous max
```

File: commands/createfixpullrequests.go [hunk 2]
```diff
@@ -126,6 +126,7 @@ func (cfp *CreateFixPullRequestsCmd) createFixVersionsMap(params *utils.FrogbotP
 					if !fixVulnerability {
 						continue
 					}
+					// FixedVersions array is sorted, and we have to take only the minimal fix version to avoid a breakage
```
Comment (sverdlov93) [inline commands/createfixpullrequests.go:129]:
```suggestion
					// Get the minimal fix version that fixes the current vulnerability. vulnerability.FixedVersions array is sorted so we take the first index.
```

File: commands/createfixpullrequests.go (hunk 1 above)
Comment (omerzi) [reply commands/createfixpullrequests.go]: done

File: commands/createfixpullrequests.go (hunk 2 above)
Comment (omerzi) [reply commands/createfixpullrequests.go]: done

File: commands/createfixpullrequests.go [hunk 3]
```diff
@@ -126,14 +126,15 @@ func (cfp *CreateFixPullRequestsCmd) createFixVersionsMap(params *utils.FrogbotP
...
 						continue
 					}
-					fixVersion := parseVersionChangeString(vulnerability.FixedVersions[0])
+					// Get the minimal fix version that fixes the current vulnerability. vulnerability.FixedVersions array is sorted, so we take the first index.
+					vulnFixVersion := parseVersionChangeString(vulnerability.FixedVersions[0])
 					fixVersionInfo, exists := fixVersionsMap[vulnerability.ImpactedPackageName]
 					if exists {
-						// Fix version for current impacted package already exists, so we need to select between the existing fix version and the current.
-						fixVersionInfo.UpdateFixVersion(fixVersion)
+						// More than one vulnerability can exist on the same impacted package. Among all possible fix versions that fix the above impacted package, we select the maximum fix version.
```
Comment (sverdlov93) [inline commands/createfixpullrequests.go:133]:
```suggestion
						// More than one vulnerability can exist on the same impacted package.
						// Among all possible fix versions that fix the above impacted package, we select the maximum fix version.
```

File: commands/createfixpullrequests.go [hunk 4]
```diff
@@ -295,7 +296,8 @@ func NewFixVersionInfo(newFixVersion string, packageType PackageType) *FixVersio
 }
 
 func (fvi *FixVersionInfo) UpdateFixVersion(newFixVersion string) {
-	if fvi.fixVersion == "" || version.NewVersion(fvi.fixVersion).AtLeast(newFixVersion) {
+	// Update fvi.fixVersion as the maximum version if found a new version that greater than the previous max
```
Comment (sverdlov93) [inline commands/createfixpullrequests.go:299]:
```suggestion
	// Update fvi.fixVersion as the maximum version if found a new version that is greater than the previous maximum version.
```

**Review by sverdlov93 — APPROVED** [review]

**Pull request merged**